src/fidesapi/routes/scan.py

```diff
@@ -0,0 +1,118 @@
+"""
+Contains all of the endpoints required to manage a scan of your resources.
+"""
+from enum import Enum
+from typing import Dict, List, Union
+
+from fastapi import APIRouter, Response, status
+from fideslang.models import Organization
+from loguru import logger as log
+from pydantic import BaseModel
+
+from fidesapi.routes.crud import get_resource
+from fidesapi.routes.util import API_PREFIX, unobscure_string
+from fidesapi.sql_models import sql_model_map
+from fidesapi.utils import errors
+from fidesctl.core.system import generate_aws_systems
+
+
+class ScanTypes(str, Enum):
+    """
+    Scan Type Enum to capture the discrete possible values
+    for a valid scan type
+    """
+
+    SYSTEM = "system"
+    DATASET = "dataset"
+
+
+class AWSConfig(BaseModel):
+    """
+    The model for the connection config for AWS
+    """
+
+    region_name: str
+    aws_secret_access_key: str
+    aws_access_key_id: str
+
+
+class OktaConfig(BaseModel):
+    """
+    The model for the connection config for Okta
+    """
+
+    okta_client_token: str
+
+
+class Scan(BaseModel):
+    """
+    Defines attributes of the scan included in a request.
+    """
+
+    config: Union[AWSConfig, OktaConfig]
+    target: str
+    type: ScanTypes
+
+
+class ScanRequestPayload(BaseModel):
+    """
+    The model for the request body housing scan information.
+    """
+
+    organization_key: str
+    scan: Scan
+
+
+router = APIRouter(tags=["Scan"], prefix=f"{API_PREFIX}/scan")
+
+
+@router.post("/", status_code=status.HTTP_201_CREATED)
+async def generate_scan(
+    scan_request_payload: ScanRequestPayload, response: Response
+) -> Dict:
+    """
+    Kicks off a generate command for fidesctl.
+
+    Starting off, this will only take 'generate system aws'
+
+    Debate is still up for if one endpoint makes sense or many, might be more
+    difficult to document etc. with all of the config options.
+
+    Currently follows the same logic as `generate_system_aws` in `system.py`
+    """
+    organization = await get_resource(
+        sql_model_map["organization"], scan_request_payload.organization_key
+    )
+    if scan_request_payload.scan.target.lower() == "aws":
+        log.info("Setting config for AWS")
+        generated_systems = generate_aws(
+            aws_config=AWSConfig(**scan_request_payload.scan.config.dict()),
+            organization=organization,
+        )
+
+    response.status_code = (
+        status.HTTP_200_OK if len(generated_systems) > 0 else status.HTTP_204_NO_CONTENT
+    )
```

Comment:
```suggestion
    if len(generated_systems):
        response.status_code =  status.HTTP_200_OK
```
Then you can just make `status.HTTP_204_NO_CONTENT` the default on L69.

Reply:
Although, thinking more about this, it's probably more conventional to make `200 OK` the default, and conditionally respond with `204 No Content`...

Reply:
I ended up removing the 204 as it didn't end up playing the role I had in mind (as not content is required to be sent in the response body apparently)